tests/spec_decode/e2e/test_correctness.py

```diff
@@ -89,6 +89,49 @@ def test_spec_decode_e2e_logical_flow(test_llm_generator, batch_size: int):
         assert actual_tokens.strip() == expected_tokens.strip()
 
 
+@pytest.mark.parametrize(
+    "common_llm_kwargs",
+    [{
+        # Skip cuda graph recording for fast test.
+        "enforce_eager": True,
+
+        # Required for spec decode.
+        "use_v2_block_manager": True
+    }])
+@pytest.mark.parametrize("per_test_common_llm_kwargs", [
+    {
+        "model": "JackFram/llama-160m",
+    },
+])
+@pytest.mark.parametrize("baseline_llm_kwargs", [{}])
+@pytest.mark.parametrize(
+    "test_llm_kwargs",
+    [
+        {
+            # use ngram prompt lookup drafter
+            "speculative_model": "[ngram]",
+            "num_speculative_tokens": 5,
+        },
+    ])
+@pytest.mark.parametrize(
+    "output_len",
+    [
+        # Use long output len for the small model test.
+        128,
+    ])
+@pytest.mark.parametrize("batch_size", [1])
+@pytest.mark.parametrize("seed", [1])
+def test_validate_ngram_prompt_lookup(baseline_llm_generator,
```

Comment:
I need to write docs still, but FYI these tests are a bit sensitive to batch size / prompts. In my investigation I find that vLLM logits can be unstable depending on some batching condition (even with torch determinism flags enabled). I haven't looked further to identify if it's framework nondeterminism, vLLM nondeterminism, or a spec decode bug. But since it's very stable for the bs/prompts that work, it's OK to rely on this for correctness.

Reply:
Got it.